controllers/extensions/addon_controller_stages.go

```diff
@@ -177,6 +178,20 @@ func (r *fetchNDeletionCheckStage) Handle(ctx context.Context) {
 	}
 	r.reqCtx.Log.V(1).Info("get addon", "generation", addon.Generation, "observedGeneration", addon.Status.ObservedGeneration)
 	r.reqCtx.UpdateCtxValue(operandValueKey, addon)
+
+	// CheckIfAddonUsedByCluster, if err, skip the deletion stage
+	if !addon.GetDeletionTimestamp().IsZero() || (addon.Spec.InstallSpec != nil && !addon.Spec.InstallSpec.GetEnabled()) {
+		used, err := CheckIfAddonUsedByCluster(ctx, r.reconciler.Client, addon.Name, r.reqCtx.Req.Namespace)
+		if err != nil {
+			r.setRequeueWithErr(err, "")
+			return
+		} else if used {
+			r.reconciler.Event(addon, corev1.EventTypeNormal, "Addon is used by some clusters",
+				"Addon is used by cluster, please check")
+			r.setRequeueAfter(time.Second, "Waiting for the cluster to end")
```

Comment:
Might this generate a lot of logs?

Reply:
Yes, messages will continue to be sent until confirmation of successful deletion. This behavior makes sense, but it does generate a lot of logs. Is there a better solution?

Reply:
Perhaps the detection interval should be increased?